pkg/cmd/kafka/create/create.go

```diff
@@ -297,6 +317,22 @@ func runCreate(opts *options) error {
 	return nil
 }
 
+func validateSize(opts *options, constants *remote.DynamicServiceConstants, conn connection.Connection) error {
+	amsType, err := accountmgmtutil.PickInstanceType(&opts.userInstanceTypes)
+	if err != nil {
+		return err
+	}
+	sizes, err := kafkautil.GetValidSizes(conn, opts.Context, opts.provider, opts.region, &amsType)
+	if err != nil {
+		return err
+	}
+	if !slices.Contains(sizes, opts.size) {
+		// TODO error message
+		return errors.New("Whatever") //opts.localizer.MustLocalizeError("")
+	}
+	return nil
```

Comment:
We also need cloud region value here to be able to match if size is allowed

Reply:
Thinking about it - would it be easier to return from API only sizes that will fit to that region?